File: day03.py
import sys
import math

input_file = sys.argv[1]

data = [line for line in open(input_file).read().splitlines()]

trees = []
slopes = [[1, 1], [3, 1], [5, 1], [7, 1], [1, 2]]
for right, down in slopes:
   i = 0; j = 0
   counter = 0
   while i in range(len(data)-down):
       if data[(i := i+down)][(j := (j+right) % len(data[0]))] == '#': counter += 1
   trees.append(counter)
print(f"{trees[1]}, {math.prod(trees)}")

File: day03.rb
input_file = ARGV[0]

data = File.readlines(input_file, chomp:true)

trees = []
slopes = [[1, 1], [3, 1], [5, 1], [7, 1], [1, 2]]


slopes.each do | right, down |
    i = j = counter = 0
    while i < (data.length - down)
        counter += 1 if data[(i = i + down)][(j = (j + right) % data[0].length())] == '#'
    end
    trees << counter
end
puts "#{trees[1]}, #{trees.reduce(:*)}"
